tests/test_runner/tests.py

```diff
@@ -404,3 +404,14 @@ def test_autoincrement_reset1(self):
     def test_autoincrement_reset2(self):
         p = Person.objects.create(first_name='Jack', last_name='Smith')
         self.assertEqual(p.pk, 1)
+
+
+class EmptyDefaultDatabaseTest(unittest.TestCase):
+    def test_empty_default_database(self):
+        """Ticket 24394 regression test"""
+        db.connections = db.ConnectionHandler(dict(default={}))
```

Comment:
Would using `mock` work here rather than adding a testing parameter to `connections_support_transactions()`?